File: angular/src/shared/Dto/Client-Audit-Visit-model.ts
export class ClientAuditVisitModel implements IClientAuditVisitModel {

    Id: number | undefined
    ProjectId: number | undefined;
    //VisitTypeId: number | undefined;
   VisitStatusId: number | undefined;
    JustifiedPersonId: number | undefined;
    TechnicalExpertId: number | undefined;

    Duration: string | undefined;
    VisitDate: string | undefined;
    StartDate: string | undefined;
    EndDate: string | undefined;
    LeadAuditorId: number | undefined;

    Auditor1Id: number | undefined;
    Auditor2Id: number | undefined;
    Auditor3Id: number | undefined;
    Auditor4Id: number | undefined;
    Auditor5Id: number | undefined;
    ReviewerId: number | undefined;
    ReviewDate: string | undefined;
    SubmissionDate: string | undefined;
    VerificationLevel: string | undefined;
    
    CreatedById: number | undefined;
    LastModifiedById: number | undefined;
    File: string | undefined;
    OrganizationId: number | undefined;
    StandardId: number | undefined;
    //IsDeleted: boolean | undefined;
    VisitLevelId

    constructor(data?: ClientAuditVisitModel) {
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }
    // VisitTypeId: string;
    // ProjectId: string;
    // id: number;

    init(data?: any) {
        if (data) {
            this.Id = data["Id"];
            this.ProjectId = data["ProjectId"];
           // this.VisitTypeId = data["VisitTypeId"];
            this.VisitStatusId = data["VisitStatusId"];
            
            this.VisitLevelId = data["VisitLevelId"];
            this.JustifiedPersonId = data["JustifiedPersonId"];
            this.TechnicalExpertId = data["TechnicalExpertId"];
            this.Duration = data["Duration"];

            this.VisitDate = data["VisitDate"];
            this.StartDate = data["StartDate"];
            this.EndDate = data["EndDate"];
            this.LeadAuditorId = data["LeadAuditorId"];
            this.Auditor1Id = data["Auditor1Id"];
            this.Auditor2Id = data["Auditor2Id"];
            this.Auditor3Id = data["Auditor3Id"];

            this.Auditor3Id = data["Auditor3Id"];
            this.Auditor4Id = data["Auditor4Id"];
            this.Auditor5Id = data["Auditor5Id"];
            this.ReviewerId=data["ReviewerId"];
            this.ReviewDate=data["ReviewDate"];
            this.SubmissionDate=data["SubmissionDate"];
            this.VerificationLevel = data["VerificationLevel"];
            this.CreatedById = data["CreatedById"];
            this.LastModifiedById = data["LastModifiedById"];
            this.File= data["File"];
            this.OrganizationId=data["OrganizationId"];
            this.StandardId=data["StandardId"];
            
            // this.IsDeleted = data["IsDeleted"];

        }
    }

    static fromJS(data: any): ClientAuditVisitModel {
        data = typeof data === 'object' ? data : {};
        let result = new ClientAuditVisitModel();
        result.init(data);
        return result;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data["Id"] = this.Id;
        data["ProjectId"] = this.ProjectId;
        //data["VisitTypeId"] = this.VisitTypeId;
       data["VisitStatusId"] = this.VisitStatusId;
        data["VisitLevelId"] = this.VisitLevelId;
        
        data["JustifiedPersonId"] = this.JustifiedPersonId;
        data["TechnicalExpertId"] = this.TechnicalExpertId;

        data["Duration"] = this.Duration;

        data["VisitDate"] = this.VisitDate;
        data["StartDate"] = this.StartDate;
        data["EndDate"] = this.EndDate;
        data["LeadAuditorId"] = this.LeadAuditorId;
        data["Auditor1Id"] = this.Auditor1Id;
        data["Auditor2Id"] = this.Auditor2Id;
        data["Auditor3Id"] = this.Auditor3Id;

        data["Auditor3Id"] = this.Auditor3Id;
        data["Auditor4Id"] = this.Auditor4Id;
        data["Auditor5Id"] = this.Auditor5Id;
        data["ReviewerId"] = this.ReviewerId;
        data["ReviewDate"] = this.ReviewDate;
        data["SubmissionDate"] = this.SubmissionDate;
        data["VerificationLevel"] = this.VerificationLevel;
        data["CreatedById"] = this.CreatedById;
        data["LastModifiedById"] = this.LastModifiedById;
        data["File"] = this.File;
        data["OrganizationId"] = this.OrganizationId;
        data["StandardId"] = this.StandardId;
        
        // data["IsDeleted"] = this.IsDeleted;

        return data;
    }

    clone(): ClientAuditVisitModel {
        const json = this.toJSON();
        let result = new ClientAuditVisitModel();
        result.init(json);
        return result;
    }
}


export interface IClientAuditVisitModel {
    Id: number;
    ProjectId: number | undefined;
    //VisitTypeId: number | undefined;
    VisitStatusId: number | undefined;
    JustifiedPersonId: number | undefined;
    TechnicalExpertId: number | undefined;
    Duration: string | undefined;
    VisitDate: string | undefined;
    StartDate: string | undefined;
    EndDate: string | undefined;
    LeadAuditorId: number | undefined;

    Auditor1Id: number | undefined;
    Auditor2Id: number | undefined;
    Auditor3Id: number | undefined;
    Auditor4Id: number | undefined;
    Auditor5Id: number | undefined;
    ReviewerId: number | undefined;
    ReviewDate: string | undefined;
    SubmissionDate: string | undefined;
    VerificationLevel: string | undefined;
    CreatedById: number | undefined;
    LastModifiedById: number | undefined;
    File: string | undefined;
    OrganizationId: number | undefined;
    StandardId: number | undefined;
    VisitLevelId: number | undefined;

}

File: Ozone.WebApi/Ozone.Application/DTOs/Projects/ClientAuditVisitModel.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;

namespace Ozone.Application.DTOs
{
    public class ClientAuditVisitModel
    {
        public long? S_No { get; set; }

        public long Id { get; set; }
        [ReadOnly(true)]
        public string Code { get; set; }
        public long? ProjectId { get; set; }
        [ReadOnly(true)]
        public string ProjectCode { get; set; }
        public long? VisitTypeId { get; set; }
        [ReadOnly(true)]
        public string VisitTypeName { get; set; }
        public long? VisitStatusId { get; set; }
        [ReadOnly(true)]
        public string VisitStatusName { get; set; }
        public long? JustifiedPersonId { get; set; }
        public string JustifiedPersonName { get; set; }
        public long? TechnicalExpertId { get; set; }
        public string TechnicalExpertName { get; set; }
        public string Duration { get; set; }
   
        public DateTime? VisitDate { get; set; }
       
        public DateTime? StartDate { get; set; }
        
        public DateTime? EndDate { get; set; }
        public long? LeadAuditorId { get; set; }
        public string LeadAuditorName { get; set; }
        public long? Auditor1Id { get; set; }
        public string Auditor1Name { get; set; }
        public long? Auditor2Id { get; set; }
        public string Auditor2Name { get; set; }
        public long? Auditor3Id { get; set; }
        public string Auditor3Name { get; set; }
        public long? Auditor4Id { get; set; }
        public string Auditor4Name { get; set; }
        public long? Auditor5Id { get; set; }
        public string Auditor5Name { get; set; }

        public string VerificationLevel { get; set; }
        public long? CreatedById { get; set; }

        public DateTime? CreatedDate { get; set; }
        public long? LastModifiedById { get; set; }
    
        public DateTime? LastModifiedDate { get; set; }
        public long? ApprovedById { get; set; }
     
        public DateTime? ApprovedDate { get; set; }
        public bool? IsDeleted { get; set; }
        public IFormFile File { get; set; }
        public string AuditPlanFilePath { get; set; }
        public string AuditPlanContentType { get; set; }
        public string StandardName { get; set; }
        
        public string Remarks { get; set; }
        public long? ReviewerId { get; set; }
        public string ReviewerName { get; set; }
        public long? StandardId { get; set; }
        public long? OrganizationId { get; set; }
        public long? ClientId{ get; set; }

        public DateTime? ReviewDate { get; set; }
       
        public DateTime? SubmisionDate { get; set; }
        public long? VisitLevelId { get; set; }
        public string VisitLevelName { get; set; }
        public string ClientName { get; set; }
        public string ClientSiteName { get; set; }
        public DateTime? ResubmitedDate { get; set; }
        public TimeSpan days { get; set; }
        public DateTime? AuditCompletedDate { get; set; }
        public string BranchName { get; set; }

        public string NaceCode { get; set; }
        public string EACode { get; set; }


    }
    public class GetPagedClientAuditVisitModel
    {
        public int TotalCount { get; set; }
        public List<ClientAuditVisitModel> ClientAuditModel { get; set; }
    }
}
